src/store.c

```diff
@@ -106,8 +106,10 @@ static void garbage_collect_dead_processes(struct ps_struct *ps_first) {
                         /* close the stream and fds */
                         ps_next->schedstat = safe_close(ps_next->schedstat);
                         ps_next->sched = safe_close(ps_next->sched);
-                        if (ps->smaps)
-                                fclose(ps->smaps);
+                        if (ps_next->smaps) {
+                                fclose(ps_next->smaps);
+                                ps_next->smaps = NULL;
```

Comment:
Likely not, but ideally we'd have some "non-NULL but invalid value" to set to `smaps` in this case to catch similar issues in the future. `(FILE *)-1` or `(FILE *)1` maybe?